relay-server/src/endpoints/minidump.rs

```diff
@@ -43,6 +50,65 @@ fn validate_minidump(data: &[u8]) -> Result<(), BadStoreRequest> {
     Ok(())
 }
 
+/// Convenience wrapper to let a decoder decode its full input into a buffer
+fn run_decoder(decoder: &mut Box<dyn Read>) -> Result<Vec<u8>, Error> {
+    let mut buffer = Vec::new();
+    decoder.read_to_end(&mut buffer)?;
+    Ok(buffer)
+}
+
+/// Creates a decoder based on the magic bytes the minidump payload
+fn decoder_from(minidump_data: Bytes) -> Option<Box<dyn Read>> {
+    if minidump_data.starts_with(b"\x1F\x8B") {
+        return Some(Box::new(GzDecoder::new(Cursor::new(minidump_data))));
+    } else if minidump_data.starts_with(b"\xFD\x37\x7A\x58\x5A\x00") {
+        return Some(Box::new(XzDecoder::new(Cursor::new(minidump_data))));
+    } else if minidump_data.starts_with(b"\x42\x5A\x68") {
+        return Some(Box::new(BzDecoder::new(Cursor::new(minidump_data))));
+    } else if minidump_data.starts_with(b"\x28\xB5\x2F\xFD") {
+        return Some(Box::new(
+            ZstdDecoder::new(Cursor::new(minidump_data)).unwrap(),
```

Comment:
Is this unwrap really safe, if so please add a comment why. Panics in Relay are currently not always handled gracefully and we should avoid them. Is logging the error and returning `None` here good enough?

Reply:
No, it is not. Thanks for reminding me. I initially thought that this was primarily due to file io errors related to loading dictionaries. Since we use dictionary-less `zstd`-decompression in this code, those shouldn't happen. 

However, any error code returned by `ZSTD_initDStream()` or `ZSTD_DCtx_loadDictionary()` will be `map_err`'ed to an `io::Error`. So I guess it is better to log and return `None`.